addons/html_editor/static/src/others/x2many_image_field.js

```diff
@@ -0,0 +1,72 @@
+import { registry } from '@web/core/registry';
+import { useService } from '@web/core/utils/hooks';
+import { ImageField, imageField } from '@web/views/fields/image/image_field';
+import { X2ManyMediaDialog } from './x2many_media_viewer';
+import { getVideoUrl } from "@html_editor/utils/url";
+
+export class X2ManyImageField extends ImageField {
+    static template = "html_editor.ImageField";
+    setup() {
+        super.setup();
+        this.orm = useService("orm");
+        this.dialog = useService("dialog");
+    }
+
+    /**
+     * New method and a new edit button is introduced here to overwrite,
+     * standard behavior of opening file input box in order to update a record.
+     */
+
+    onFileEdit(ev) {
+        const isVideo = this.props.record.data.video_url;
+        let mediaEl;
+        if (isVideo) {
+            mediaEl = document.createElement("img");
+            mediaEl.dataset.src = this.props.record.data.video_url;
+        }
+        this.dialog.add(X2ManyMediaDialog, {
+            noDocuments: true,
+            noIcons: true,
+            media: mediaEl,
+            activeTab: isVideo ? "VIDEOS" : "IMAGES",
+            save: (el) => {}, // Simple rebound to fake its execution
```

Comment:
This feels wrong to me. See `X2ManyMediaDialog`.

Reply:
Taken the reference from: `community/addons/website_sale/static/src/js/website_sale.editor.js`

Yes it looks sketchy, but this is the only way to let my custom prop methods to run for saving image and videos.